libs/vkd3d/command.c

```diff
@@ -18059,6 +18071,63 @@ static VkResult d3d12_command_queue_submit_split_locked(struct d3d12_device *dev
 	return VK_SUCCESS;
 }
 
+static void d3d12_command_queue_wait_staggered_submission(struct d3d12_command_queue *command_queue)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &command_queue->device->vk_procs;
+    VkSemaphoreWaitInfo wait_info;
+    VkResult vr;
+
+    if (!command_queue->last_submission_timeline_value)
+        return;
+
+    memset(&wait_info, 0, sizeof(wait_info));
+    wait_info.sType = VK_STRUCTURE_TYPE_SEMAPHORE_WAIT_INFO;
+    wait_info.semaphoreCount = 1;
+    wait_info.pSemaphores = &command_queue->vkd3d_queue->submission_timeline;
+    wait_info.pValues = &command_queue->last_submission_timeline_value;
```

Comment:
Did we test if a full wait-for-idle is really necessary?

Reply:
I actually did experiment with that last night and it did work, yes.